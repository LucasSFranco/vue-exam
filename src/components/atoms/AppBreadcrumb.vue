<template>
  <div class="breadcrumb">
    <template
      v-for="(item, order) in items"
      :key="order"
    >
      <span v-if="order !== 0">
        <Icon icon="chevron-right" />
        {{item}}
      </span>
      <span v-else>
        {{item}}
      </span>
    </template>
  </div>
</template>

<script>

export default {
  name: 'AppBreadcrumb',
  props: {
    items: {
      required: true,
      type: Array,
    }
  }
}

</script>

<style lang="scss" scoped>

  .breadcrumb {
    span {
      color: var(--text-secondary);
      font-size: .75rem;
      font-family: 'Roboto', sans-serif;
      text-transform: uppercase;

      svg {
        color: var(--text-muted);
        margin: 0 .25rem;
      }
    }
  }

</style>
