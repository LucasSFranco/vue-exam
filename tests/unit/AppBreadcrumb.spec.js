import { shallowMount } from '@vue/test-utils'

import AppBreadcrumb from '@/components/atoms/AppBreadcrumb.vue'

describe('Breadcrumb', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(AppBreadcrumb, {
      props: {
        items: ['item-1', 'item-2', '']
      },
      global: {
        stubs: {
          Icon: true
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
