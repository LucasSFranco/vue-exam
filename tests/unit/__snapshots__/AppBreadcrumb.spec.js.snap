// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb matches snapshot 1`] = `
<div class="breadcrumb"><span>item-1</span><span><icon-stub icon="chevron-right"></icon-stub> item-2</span>
  <!--v-if-->
</div>
`;
